modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java

```diff
@@ -1712,6 +1717,80 @@ public void testSearch() throws Exception {
 		_assertSearch("[Pepsi Cola]");
 	}
 
+	@Test
+	public void testStagingAware() throws Exception {
+		_updateConfigurationJSON(
+			"queryConfiguration", JSONUtil.put("applyIndexerClauses", false));
+
+		_enableLocalStaging();
+
+		Group stagingGroup = _group.getStagingGroup();
+
+		_journalArticles.add(
+			JournalTestUtil.addArticle(
+				stagingGroup.getGroupId(), 0,
+				PortalUtil.getClassNameId(JournalArticle.class),
+				HashMapBuilder.put(
+					LocaleUtil.US, "Staged"
+				).build(),
+				null,
+				HashMapBuilder.put(
+					LocaleUtil.US, StringPool.BLANK
+				).build(),
+				LocaleUtil.getSiteDefault(), false, true, _serviceContext));
+
+		_updateElementInstancesJSON(
+			new Object[] {
+				HashMapBuilder.<String, Object>put(
+					"boost", 1
+				).put(
+					"fields", new String[] {"title_${context.language_id}^2"}
+				).put(
+					"fuzziness", "AUTO"
+				).put(
+					"keywords", "${keywords}"
+				).put(
+					"minimum_should_match", 0
+				).put(
+					"operator", "or"
+				).put(
+					"slop", 0
+				).put(
+					"type", "best_fields"
+				).build()
+			},
+			new String[] {"Text Match Over Multiple Fields"});
+
+		_keywords = "Staged";
+
+		_assertSearch("[Staged]");
+
+		_updateElementInstancesJSON(
+			new Object[] {
+				HashMapBuilder.<String, Object>put(
+					"boost", 1
+				).put(
+					"fields", new String[] {"title_${context.language_id}^2"}
+				).put(
+					"fuzziness", "AUTO"
+				).put(
+					"keywords", "${keywords}"
+				).put(
+					"minimum_should_match", 0
+				).put(
+					"operator", "or"
+				).put(
+					"slop", 0
+				).put(
+					"type", "best_fields"
+				).build(),
+				null
+			},
+			new String[] {"Text Match Over Multiple Fields", "Staging Aware"});
+
+		_assertSearch("[]");
```

Comment:
Here and a few other places, we want to make sure that the filtering doesn't just remove results but also still allows some through. Add a unstaged JournalArticle to this test that we would still expect to be returned in this last case.